### PR TITLE
Issue98 fix pairing

### LINKED ISSUE #98: fix pairing

### PR DESCRIPTION
**Pull request by kmrozewski**
- find at least triple slots after pair matching (was before matching pairs)
- match slots times by id
- find pairs when slot availability equals to "available" and "maybe"
- added unit tests
- resolved conflicts

Fixes #98

### CODE REVIEW

File: src/test/java/resourceTests/PairResourceTests/MockDataUtil.java [hunk 1]
```diff
@@ -67,25 +69,38 @@ public static String convertDateToString(Date date) {
         return dateFormat.format(date);
     }
 
-    public static List<Slot> createSlotToSlotTime(List<SlotTime> slotTime, Person person, Date date, AvailabilityType availabilityType) {
+    public static List<Slot> createSlotsToSlotTimes(List<SlotTime> slotsTimes, Person person, Date date, AvailabilityType availabilityType) {
```
Comment (drauf) [inline src/test/java/resourceTests/PairResourceTests/MockDataUtil.java:73]: maybe name it `createSlotsForSlotTimes`?

Comment (RogowskaMalgorzata) [reply src/test/java/resourceTests/PairResourceTests/MockDataUtil.java:73]: At first it had this name but I've changed it in my merge later, sorry :<

File: src/main/java/services/PairFinder.java [hunk 2]
```diff
@@ -212,14 +222,22 @@ private Person getPersonsListElement(List<Person> persons, Long personId) {
     }
 
     private List<Slot> findPairsInSlots(List<Slot> remainingSlots, List<Slot> searchSlots) {
-        return remainingSlots
+        List<Slot> foundSlots = new ArrayList<>();
+
+        remainingSlots
                 .stream()
-                .filter(rs -> searchSlots
```
Comment (rlewan) [inline src/main/java/services/PairFinder.java:217]: Please don't use abbreviations in variable names.

File: src/main/java/services/PairFinder.java [hunk 3]
```diff
@@ -13,13 +15,21 @@
...
+        return filterTriples(groupedPersons);
+    }
 
-        return transformPairs(pairsIntoPersons);
+    private List<Person> filterTriples(List<Person> persons) {
+        return persons
+                .stream()
+                .peek(person -> person.setSlotList(findTriplesInSlots(person.getSlotList())))
+                .filter(person -> person.getSlotList().size() > 0)
+                .collect(Collectors.toCollection(ArrayList::new));
```
Comment (rlewan) [inline src/main/java/services/PairFinder.java]: You can use `Collectors.toList()` for simplicity and readability.

File: src/main/java/services/PairFinder.java (hunk 2 above)
Comment (kmrozewski) [reply src/main/java/services/PairFinder.java:217]: done

**Pull request merged**